Anagram/Anagram.cs

```diff
@@ -1,16 +1,43 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
 
 namespace Anagram
 {
     public class AnagramSelector
     {
+        int[] hashOfWord1 = new int[257];
+        public void MapEachWord(string word1)
+        {
+            Array.Clear(hashOfWord1, 0, hashOfWord1.Length);
+            for (int iterateWord1 = 0; iterateWord1 < word1.Length; iterateWord1++)
+            {
+                hashOfWord1[word1[iterateWord1] - 'a']++;
```

Comment:
If the input string contains numbers or punctuations, this subtraction from 'a' (the index into the hash) will be negative. That's because those ASCII codes are less than 'a'.
BTW, why subtract from 'a'? You have an array that can hold the entire ASCII range.
If you need to correct this, you have to fix at multiple places in the code. That indicates some duplication. See if you can identify that and make it into one function.
Also, add test cases for the first three examples shown in WikiPedia and see if any change to your function is needed.